zerver/models.py

```diff
@@ -773,6 +773,45 @@ class RealmFilter(models.Model):
     class Meta:
         unique_together = ("realm", "pattern")
 
+    def clean(self) -> None:
+        # Django's `full_clean` calls `clean_fields` followed by `clean` and
+        # stores all ValidationErrors from all stages to return as JSON. Thus,
+        # these validations are in addition to field specific validations.
+
+        # extract substitutions
+        try:
+            pattern = re.compile(self.pattern)
+        except Exception:
+            # regex validation errors would be caught by full_clean()
+            # so we can safely just return here.
+            return
```

Comment:
This feels unsafe; can we just raise a validation error here?

Reply:
Or maybe better, just remove the field-level validator and do the clean check here?